kvmagent/kvmagent/test/test_ceph_storage_plugin_luks_convert.py

```
@@ -0,0 +1,175 @@
+    zstacklib_pkg = types.ModuleType("zstacklib")
+    utils_pkg = types.ModuleType("zstacklib.utils")
+    zstacklib_pkg.utils = utils_pkg
+    sys.modules["zstacklib"] = zstacklib_pkg
+    sys.modules["zstacklib.utils"] = utils_pkg
+
+    class AttrDict(dict):
+        def __getattr__(self, item):
+            try:
+                return self[item]
+            except KeyError:
+                raise AttributeError(item)
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)